src/core/lib/iomgr/work_serializer.cc

```diff
@@ -60,16 +34,41 @@ class WorkSerializer::WorkSerializerImpl : public Orphanable {
   void Orphan() override;
 
  private:
+  struct CallbackWrapper {
+    CallbackWrapper(std::function<void()> cb,
+                    const grpc_core::DebugLocation& loc)
+        : callback(std::move(cb)), location(loc) {}
+
+    MultiProducerSingleConsumerQueue::Node mpscq_node;
+    const std::function<void()> callback;
+    const DebugLocation location;
+  };
+
   // Callers of DrainQueueOwned should make sure to grab the lock on the
   // workserializer with
-  //   prev_ref_pair = refs_.fetch_add(MakeRefPair(1, 1),
-  //   std::memory_order_acq_rel);
+  //
+  //   prev_ref_pair =
+  //     refs_.fetch_add(MakeRefPair(1, 1), std::memory_order_acq_rel);
+  //
   // and only invoke DrainQueueOwned() if there was previously no owner. Note
   // that the queue size is also incremented as part of the fetch_add to allow
   // the callers to add a callback to the queue if another thread already holds
   // the lock to the work serializer.
   void DrainQueueOwned();
 
+  // First 16 bits indicate ownership of the WorkSerializer, next 48 bits are
+  // queue size (i.e., refs).
+  uint64_t MakeRefPair(uint16_t owners, uint64_t size) {
```

Comment:
These methods can be `static`, since they don't need to access any data members.